File: programmers/lv0/코딩기초트레이닝/연산과조건문/홀짝에따라다른값반환하기/Solution.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    const bool isEven = n % 2 == 0;
    int start = 1;
    if (isEven) {
        start = 2;
    }
    for (int i = start; i <= n; i += 2) {
        if (isEven) {
            answer += i * i;
        } else {
            answer += i;
        }
    }
    return answer;
}


int main(int argc, char *argv[]) {
    cout << solution(7) << endl; // 16
    cout << solution(10) << endl; // 220
}

File: programmers/lv0/코딩기초트레이닝/연산과조건문/홀짝에따라다른값반환하기/Solution.java
package lv0.코딩기초트레이닝.연산과조건문.홀짝에따라다른값반환하기;

class Solution {
    public int solution(int n) {
        int answer = 0;
        final boolean isEven = n % 2 == 0;
        int start = isEven ? 2 : 1;
        for (int i = start; i <= n; i += 2) {
            answer += isEven ? i * i : i;
        }
        return answer;
    }
}